### PR TITLE
LPS-115073 Fixed header does not show in legacy permission modals

### PR DESCRIPTION
**Pull request by markocikos**
Hey @wincent ,

This is technically a regression bug, caused by https://github.com/brianchandotcom/liferay-portal/pull/88948/commits/20bfae4195144950f57566b56493ee3743477ff1. More accurately, we exposed an ancient bug, where `afterIframeLoaded` util executed long before iframe loaded.

Test Plan:
1. Open one of the legacy permission modals. For example, Content & Data >  Bookmarks > Folder item > Permissions.
1. Reduce browser height so that table overflows.
1. Scroll down
- before the fix: fixed header does not appear
- after the fix: fixed header appears and behaves correctly

screen

Please review the code.

Thank you!

/cc @jbalsas

### CODE REVIEW

File: modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/util_window.js [hunk 1]
```diff
@@ -166,6 +164,12 @@ AUI.add(
 
 							var iframeNode = modal.iframe.node;
 
+							var iframeElement = iframeNode.getDOM();
+
+							iframeElement.onload = function () {
+								Util.afterIframeLoaded(event);
+							};
+
```
Comment (wincent) [inline modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/util_window.js:172]: I'm wondering if there is a race condition possible here. Previously, we assumed loading was done if `event.doc` was truthy; whereas now we are assuming it is *not* done and the `load` event won't have fired until we get to this line.

I don't have enough context on this to reason about whether that assumption is valid. What do you think? If it is a risk, maybe sticking it in an `else` branch would mitigate it.

**Comment by wincent** [comment]
ci:test:sf

**Comment by wincent** [comment]
ci:test:relevant

**Comment by wincent** [comment]
ci:test:bundle

File: modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/util_window.js (hunk 1 above)
Comment (markocikos) [reply modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/util_window.js:172]: `event.doc` contains iframe `document` object. I am not sure if there is a deeper purpose behind that check, a case where this is not set, or if it's just a simple sanity check. But, `afterIframeLoaded` does depends on it, so I'll add a commit to preserve the check.

I don't think we have a race condition. The order of execution should be:
1. `popupReady`. We define `onload` listener.
1. iframe starts loading, the code in bottom.jspf is a part of it. We define listener on `modalIframeLoaded`.
1. iframe finishes loading.  `onload` listener triggers and fires `modalIframeLoaded`
1. on `modalIframeLoaded` listener triggers, setting up the header.

Easy!

Comment (wincent) [reply modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/util_window.js]: Ok, thanks for the analysis.


trust

**Comment by wincent** [comment]
ci:test:sf

**Comment by wincent** [comment]
ci:test:relevant

**Comment by wincent** [comment]
ci:test:bundle

**Comment by wincent** [comment]
Works fine:

Screenshot 2020-06-09 -104708-OgGBfWhX@2x

**Comment by wincent** [comment]
ci:forward

**Pull request closed without merging**